src/app/portlets/dot-templates/dot-template-list/dot-template-list.component.spec.ts

```diff
@@ -83,40 +218,344 @@ describe('DotTemplateListComponent', () => {
+    it('should set Action Header options correctly', () => {
+        const model: ButtonModel[] = dotListingDataTable.actionHeaderOptions.primary.model;
+        expect(model[0].label).toEqual('Template Designer');
+        expect(model[1].label).toEqual('Advanced Template');
+        model[0].command();
+        expect(dotRouterService.gotoPortlet).toHaveBeenCalledWith('/templates/new/designer');
+        model[1].command();
+        expect(dotRouterService.gotoPortlet).toHaveBeenCalledWith('/templates/new/advanced');
     });
 
-    xit('should render passed template', () => {
-        // TODO: Find a way to test the ng-template.
+    describe('row actions', () => {
+        it('should set actions to publish template', () => {
+            const publishRow: DotActionMenuButtonComponent = rowActions[0].componentInstance;
+            expect(publishRow.actions.length).toEqual(6);
+            checkBasicOptions(publishRow);
+            expect(publishRow.actions[4].menuItem.label).toEqual('Unpublish');
+            expect(publishRow.actions[5].menuItem.label).toEqual('Copy');
+        });
+
+        it('should set actions to unPublish template', () => {
+            const unpublish: DotActionMenuButtonComponent = rowActions[2].componentInstance;
+            checkBasicOptions(unpublish);
+            expect(unpublish.actions[4].menuItem.label).toEqual('Archive');
+            expect(unpublish.actions[5].menuItem.label).toEqual('Copy');
+        });
+
+        it('should set actions to archived template', () => {
+            const archived: DotActionMenuButtonComponent = rowActions[3].componentInstance;
+            expect(archived.actions.length).toEqual(2);
+            expect(archived.actions[0].menuItem.label).toEqual('Unarchive');
+            expect(archived.actions[1].menuItem.label).toEqual('Delete');
+        });
+
+        it('should set actions to locked template', () => {
+            const locked: DotActionMenuButtonComponent = rowActions[1].componentInstance;
+            expect(locked.actions.length).toEqual(7);
+            checkBasicOptions(locked);
+            expect(locked.actions[5].menuItem.label).toEqual('Unlock');
+            expect(locked.actions[6].menuItem.label).toEqual('Copy');
+        });
+
+        it('should hide push-publish and Add to Bundle actions', () => {
+            let activatedRoute: ActivatedRoute;
+            activatedRoute = TestBed.inject(ActivatedRoute);
+            spyOnProperty(activatedRoute, 'data', 'get').and.returnValue(
+                of({
+                    dotTemplateListResolverData: [templatesMock, false, false]
+                })
+            );
+            comp.ngOnInit();
+            fixture.detectChanges();
+            const publishRow: DotActionMenuButtonComponent = rowActions[0].componentInstance;
+            expect(publishRow.actions.length).toEqual(4);
+            expect(publishRow.actions[0].menuItem.label).toEqual('Edit');
```

Comment:
```suggestion
            expect(publishRow.actions).toEqual([{label: 'XXXX', command: jasmine.any(Function)}, {...}]);
```

So you don't have to have tons of expects harder to debug if one fail.